protos/core.proto

```diff
@@ -289,99 +289,375 @@ enum SupportedFeatures {
+// configuration file.
+message WorkflowSettingsConfig {
+  // The desired dataform core version to compile against.
+  string dataform_core_version = 1;
+
+  // Required. The default database (Google Cloud project ID).
+  string default_database = 2;
+
+  // Required. The default schema (BigQuery dataset ID).
+  string default_schema = 3;
+
+  // Required. The default BigQuery location to use. For more information on
+  // BigQuery locations, see https://cloud.google.com/bigquery/docs/locations.
+  string default_location = 4;
+
+  // Required. The default schema (BigQuery dataset ID) for assertions.
+  string assertion_schema = 5;
+
+  // Optional. User-defined variables that are made available to project code
+  // during compilation. An object containing a list of "key": value pairs.
+  // Example: `{ "name": "wrench", "mass": "1.3kg", "count": "3" }`.
+  map<string, string> vars = 6;
+
+  // Optional. The suffix that should be appended to all database (Google Cloud
+  // project ID) names.
+  string database_suffix = 7;
+
+  // Optional. The suffix that should be appended to all schema (BigQuery
+  // dataset ID) names.
+  string schema_suffix = 8;
+
+  // Optional. The prefix that should be prepended to all table names.
+  string table_prefix = 9;
+}
+
+// Action configs defines the contents of `actions.yaml` configuration files.
+// TODO(ekrekr): consolidate these configuration options in the JS API.
 message ActionConfigs {
   repeated ActionConfig actions = 1;
 }
 
+// Action config defines the configuration properties of actions.
 message ActionConfig {
-  // Target is a unique action identifier.
-  Target target = 1;
+  message ColumnDescriptor {
+    // The identifier for the column, using multiple parts for nested records.
+    repeated string path = 1;
 
-  // A list of user-defined tags with which the action should be labeled.
-  repeated string tags = 2;
+    // A text description of the column.
+    string description = 2;
 
-  // Targets of actions that this action is dependent on.
-  repeated Target dependency_targets = 3;
+    // A list of BigQuery policy tags that will be applied to the column.
+    repeated string bigquery_policy_tags = 3;
+  }
 
-  // Filename to load the action contents from.
-  string file_name = 4;
+  message TableConfig {
+    // Target is the unique action identifier.
+    Target target = 1;
+
+    // Targets of actions that this action is dependent on.
+    repeated Target dependency_targets = 2;
+
+    // Path to the source file that the contents of the action is loaded from.
+    string filename = 3;
+
+    // A list of user-defined tags with which the action should be labeled.
+    repeated string tags = 4;
+
+    // If set to true, this action will not be executed. However, the action can
+    // still be depended upon. Useful for temporarily turning off broken
+    // actions.
+    bool disabled = 5;
+
+    // The SQL query.
+    string query = 6;
```

Comment:
surely this should be in a `oneof` with `filename`?

Reply:
Let's just not support the query field until we know that's something we want to do long-term